client/state/sites/plans/actions.js

```diff
@@ -31,39 +28,13 @@ import wpcom from 'lib/wp';
  * @returns {Function} a promise that will resolve once updating is completed
  */
 export function cancelSitePlanTrial( siteId, planId ) {
-	return ( dispatch ) => {
-		dispatch( {
-			type: SITE_PLANS_TRIAL_CANCEL,
-			siteId
-		} );
-
-		return new Promise( ( resolve, reject ) => {
-			wpcom.undocumented().cancelPlanTrial( planId, ( error, data ) => {
-				if ( data && data.success ) {
-					dispatch( {
-						type: SITE_PLANS_TRIAL_CANCEL_COMPLETED,
-						siteId,
-						plans: map( data.plans, createSitePlanObject )
-					} );
-
-					resolve();
-				} else {
-					debug( 'Canceling site plan trial failed: ', error );
-
-					const errorMessage = error.message || i18n.translate( 'There was a problem canceling the plan trial. Please try again later or contact support.' );
-
-					dispatch( {
-						type: SITE_PLANS_TRIAL_CANCEL_FAILED,
-						siteId,
-						error: errorMessage
-					} );
-
-					reject( errorMessage );
-				}
-			} );
-		} );
+	return {
+		type: SITE_PLANS_TRIAL_CANCEL,
+		siteId,
+		planId
 	}
 }
+registerConnection( SITE_PLANS_TRIAL_CANCEL, connectCancelSitePlanTrial, { offlineQueue: false } );
```

Comment:
I think the call to `registerConnection` here would be better placed within `connections.js` so we can keep `actions.js` really simple and uniform, and the queue options are declared where they can be better cross-referenced against the async code.